### ISSUE TITLE
1.17+ YoFPS/Vector-Shader/Oceano 3.0.1: The shadows under the entities are not correct.

### ISSUE THREAD

**Issue by Lolothepro** [IrisShaders/Iris#1427]
### What happened?

1.17+ YoFPS: The shadows under the entities are not correct.

### Screenshots

![2022-05-15_12 26 49](https://user-images.githubusercontent.com/68275908/168468212-76863250-e915-4d55-8958-ba2da933c783.png)

### Relevant log output

```shell
[12:26:17] [main/INFO]: Loading Minecraft 1.17.1 with Fabric Loader 0.14.5
[12:26:17] [main/WARN]: Warnings were found!
 - Mod 'Fabric Rendering Fluids (v1)' (fabric-rendering-fluids-v1) 0.1.13+a02b446313 recommends any version of fabric-textures-v0, which is missing!
	 - You should install any version of fabric-textures-v0 for the optimal experience.
[12:26:17] [main/INFO]: Loading 14 mods:
	- fabric-api-base 0.4.0+cf39a74318 via iris
	- fabric-command-api-v1 1.1.4+cf39a74318 via iris
	- fabric-key-binding-api-v1 1.0.6+2a2bb57318 via iris
	- fabric-rendering-data-attachment-v1 0.1.5+a02b446313 via sodium
	- fabric-rendering-fluids-v1 0.1.13+a02b446313 via sodium
	- fabric-resource-loader-v0 0.4.10+f09604ce18 via iris
	- fabricloader 0.14.5
	- iris 1.2.4-build.37
	- java 17
	- minecraft 1.17.1
	- org_anarres_jcpp 1.4.14 via iris
	- org_joml_joml 1.10.2 via sodium
	- org_slf4j_slf4j-api 1.7.12 via iris
	- sodium 0.3.4+build.13
[12:26:17] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/home/loic/.local/share/PolyMC/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[12:26:17] [main/INFO]: Compatibility level set to JAVA_16
[12:26:18] [main/INFO]: Loaded configuration file for Sodium: 29 options available, 1 override(s) found
[12:26:18] [main/WARN]: Force-disabling mixin 'features.sky.MixinWorldRenderer' as rule 'mixin.features.sky' (added by mods [iris]) disables it and children
[12:26:18] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[12:26:18] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[12:26:22] [Render thread/WARN]: Method overwrite conflict for method_22920 in sodium.mixins.json:features.buffer_builder.intrinsics.MixinBufferBuilder from mod sodium, previously written by net.coderbot.iris.mixin.vertices.block_rendering.MixinBufferBuilder_SeparateAo. Skipping method.
[12:26:23] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[12:26:23] [Render thread/INFO]: Setting user: Lolothepro29
[12:26:23] [Render thread/INFO]: Backend library: LWJGL version 3.2.2 build 10
[12:26:23] [Render thread/WARN]: OpenGL debug context activated.
[12:26:24] [Render thread/INFO]: [GL] Using KHR_debug for error logging.
[12:26:24] [Render thread/INFO]: Debug functionality is enabled, logging will be more verbose!
[12:26:24] [Render thread/INFO]: null:1:1: error: Unknown preprocessor directive final
[12:26:24] [Render thread/INFO]: null:4:1: error: Unknown preprocessor directive subscreen
[12:26:24] [Render thread/WARN]: Unable to resolve shader pack option menu element "" defined in shaders.properties
[12:26:24] [Render thread/INFO]: Profile: Custom (+1 option changed by user)
[12:26:24] [Render thread/INFO]: Using shaderpack: YoFPS+High+(Chocapic13+Edit).zip
[12:26:25] [Render thread/INFO]: Hardware information:
[12:26:25] [Render thread/INFO]: CPU: 12x AMD Ryzen 5 1600 Six-Core Processor
[12:26:25] [Render thread/INFO]: GPU: NVIDIA GeForce GTX 1070/PCIe/SSE2 (Supports OpenGL 3.2.0 NVIDIA 515.43.04)
[12:26:25] [Render thread/INFO]: OS: Linux (5.17.0-7.1-liquorix-amd64)
[12:26:26] [Render thread/WARN]: Method overwrite conflict for getNativeMemoryUsage in sodium.mixins.json:features.debug.MixinDebugHud from mod sodium, previously written by net.coderbot.iris.mixin.MixinDebugScreenOverlay. Skipping method.
[12:26:26] [Render thread/INFO]: Narrator library successfully loaded
[12:26:26] [Render thread/INFO]: Reloading ResourceManager: Default, Fabric Mods (Fabric API Base, Fabric Command API (v1), Fabric Key Binding API (v1), Fabric Rendering Data Attachment (v1), Fabric Rendering Fluids (v1), Fabric Resource Loader (v0), Fabric Loader, Iris, Sodium)
[12:26:28] [Render thread/INFO]: OpenAL initialized.
[12:26:28] [Render thread/INFO]: Sound engine started
[12:26:28] [Render thread/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
[12:26:28] [Render thread/INFO]: Created: 256x128x4 minecraft:textures/atlas/signs.png-atlas
[12:26:28] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[12:26:28] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[12:26:28] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[12:26:28] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[12:26:28] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[12:26:30] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/particles.png-atlas
[12:26:30] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
[12:26:30] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[12:26:35] [Render thread/WARN]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[12:26:35] [Render thread/WARN]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[12:26:35] [Render thread/WARN]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[12:26:35] [Render thread/WARN]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[12:26:35] [Render thread/WARN]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[12:26:35] [Render thread/INFO]: Reloading ResourceManager: Default
[12:26:35] [Render thread/INFO]: Loaded 7 recipes
[12:26:35] [Render thread/INFO]: Loaded 1137 advancements
[12:26:37] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[12:26:37] [Server thread/INFO]: Starting integrated minecraft server version 1.17.1
[12:26:37] [Server thread/INFO]: Generating keypair
[12:26:37] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[12:26:38] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 0%
[12:26:41] [Render thread/INFO]: Preparing spawn area: 24%
[12:26:42] [Render thread/INFO]: Preparing spawn area: 62%
[12:26:42] [Render thread/INFO]: Preparing spawn area: 84%
[12:26:43] [Render thread/INFO]: Preparing spawn area: 89%
[12:26:43] [Render thread/INFO]: Time elapsed: 5903 ms
[12:26:43] [Server thread/INFO]: Changing view distance to 11, from 10
[12:26:44] [Server thread/INFO]: Lolothepro29[local:E:2a6dd32f] logged in with entity id 184 at (265.79888144829033, 70.86859856141189, 174.4551164693027)
[12:26:44] [Server thread/INFO]: Lolothepro29 joined the game
[12:26:44] [Render thread/INFO]: Reloading pipeline on dimension change: null => OVERWORLD
[12:26:44] [Render thread/INFO]: Creating pipeline for dimension OVERWORLD
[12:26:44] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[12:26:45] [Render thread/INFO]: Started 6 worker threads
[12:26:45] [Server thread/INFO]: Saving and pausing game...
[12:26:45] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[12:26:46] [Render thread/INFO]: Loaded 0 advancements
[12:26:46] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[12:26:46] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[12:26:49] [Render thread/INFO]: [CHAT] Saved screenshot as 2022-05-15_12.26.49.png
[12:26:49] [Server thread/INFO]: Saving and pausing game...
[12:26:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[12:26:50] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[12:26:50] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[12:26:50] [Server thread/INFO]: Lolothepro29 lost connection: Disconnected
[12:26:50] [Server thread/INFO]: Lolothepro29 left the game
[12:26:50] [Server thread/INFO]: Stopping singleplayer server as player logged out
[12:26:50] [Server thread/INFO]: Stopping server
[12:26:50] [Server thread/INFO]: Saving players
[12:26:50] [Server thread/INFO]: Saving worlds
[12:26:50] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[12:26:50] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[12:26:50] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[12:26:50] [Server thread/INFO]: ThreadedAnvilChunkStorage (New World): All chunks are saved
[12:26:50] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[12:26:50] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[12:26:50] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[12:26:50] [Render thread/INFO]: Stopping worker threads
[12:26:51] [Render thread/INFO]: Stopping!
```

### Minecraft Version

Minecraft 1.18.2

### Iris Version

iris-mc-1.18.2-1.2.4

### Sodium Version

sodium 0.4.1

### Operating System

Ubuntu 22.04

### What is your GPU?

GTX 1070

### Java Version

Java 17

### Additional context

Iris 1.1.4

**Comment by Lolothepro** [comment]
Also occurs with [Vector-Shader](https://github.com/WoMspace/Vector-Shader)

**Comment by IMS212** [comment]
Fixed in 1.6.5.
